evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -1428,3 +1428,59 @@ def test_time_index_cannot_be_none(time_series_regression_pipeline_class):
                 }
             }
         )
+
+
+@pytest.mark.parametrize(
+    "gap,reset_index,freq",
+    [
+        (0, False, "1D"),
+        (0, True, "3D"),
+        (1, False, "1D"),
+        (1, True, "1D"),
+        (5, False, "1D"),
+        (5, True, "1D"),
+        (5, False, None),
+    ],
+)
+@pytest.mark.parametrize(
+    "pipeline_class,estimator_name",
+    [
+        (TimeSeriesRegressionPipeline, "Random Forest Regressor"),
+        (TimeSeriesBinaryClassificationPipeline, "Logistic Regression Classifier"),
+        (TimeSeriesMulticlassClassificationPipeline, "Logistic Regression Classifier"),
+    ],
+)
+def test_noninferrable_data(pipeline_class, estimator_name, gap, reset_index, freq):
```

Comment:
is there test coverage for when the gaps should raise an error/return false?

Reply:
Are you referring to line [1479](https://github.com/alteryx/evalml/pull/3160/files#diff-1b99a3819e82323aca1e02c75772961e13269542eccb070128773fb5bcc5f024R1479) below? It covers uninferrable holdout data.
The other `ValueError` in `_are_datasets_separated_by_gap_time_index` is covered by frequencies of 1 in this test, where the output frequency is `D` or `M` instead of `1D` or `1M`.
I also went ahead and added to `test_time_series_pipeline_validates_holdout_data` to cover the frequency aspect of the error!